File: android/app/src/main/java/com/speakpro/features/home/HomeStylePreference.kt
package com.speakpro.features.home

import android.content.Context
import android.content.SharedPreferences
import androidx.compose.runtime.mutableStateOf

/**
 * 首页风格偏好（dev 彩蛋：连点 masthead 的 No.XXX 3 下打开切换 sheet）。
 *
 * 对应设计 speakpro/components/{HomeEditorial,HomeVariants}.jsx：
 *   - Editorial 风格 × 3 种模考卡片（Full / Ticket / Diagram）
 *   - Minimal / Dashboard 目前占位（UI 未实现，敬请期待）
 */
enum class HomeStyle(val label: String, val subtitle: String, val available: Boolean) {
    EDITORIAL_FULL("编辑风 · 完整卡", "深底 hero + 3 项统计", true),
    EDITORIAL_TICKET("编辑风 · 票根卡", "左侧 sienna 票根 + 虚线分割", true),
    EDITORIAL_DIAGRAM("编辑风 · 示意图", "P1/P2/P3 schematic", true),
    MINIMAL("极简风", "大量留白 · 单列信息", false),
    DASHBOARD("仪表盘风", "圆环评分 + 多图表", false);

    companion object {
        val default = EDITORIAL_FULL
        fun fromName(name: String?): HomeStyle = entries.firstOrNull { it.name == name } ?: default
    }
}

object HomeStylePreference {

    private const val PREF = "speakpro_home_style"
    private const val KEY = "home_style"

    private lateinit var prefs: SharedPreferences

    /** 可在 Composable 里 collectAsState 的响应式状态 */
    val current = mutableStateOf(HomeStyle.default)

    fun init(context: Context) {
        prefs = context.getSharedPreferences(PREF, Context.MODE_PRIVATE)
        current.value = HomeStyle.fromName(prefs.getString(KEY, null))
    }

    fun set(style: HomeStyle) {
        current.value = style
        prefs.edit().putString(KEY, style.name).apply()
    }
}

File: ios/SpeakPro/Features/Home/HomeStylePreference.swift
import Foundation
import SwiftUI

/// 首页风格偏好 —— dev 彩蛋：连点 masthead 的 No.XXX 3 下可切换
///
/// 对应 Android `features/home/HomeStylePreference.kt`，两端行为一致：
///   Editorial 风 × 3 种模考卡（Full / Ticket / Diagram）+ Minimal / Dashboard 占位
enum HomeStyle: String, CaseIterable, Identifiable {
    case editorialFull = "EDITORIAL_FULL"
    case editorialTicket = "EDITORIAL_TICKET"
    case editorialDiagram = "EDITORIAL_DIAGRAM"
    case minimal = "MINIMAL"
    case dashboard = "DASHBOARD"

    var id: String { rawValue }

    var label: String {
        switch self {
        case .editorialFull:    return "编辑风 · 完整卡"
        case .editorialTicket:  return "编辑风 · 票根卡"
        case .editorialDiagram: return "编辑风 · 示意图"
        case .minimal:          return "极简风"
        case .dashboard:        return "仪表盘风"
        }
    }

    var subtitle: String {
        switch self {
        case .editorialFull:    return "深底 hero + 3 项统计"
        case .editorialTicket:  return "左侧 sienna 票根 + 虚线分割"
        case .editorialDiagram: return "P1/P2/P3 schematic"
        case .minimal:          return "大量留白 · 单列信息"
        case .dashboard:        return "圆环评分 + 多图表"
        }
    }

    var available: Bool {
        switch self {
        case .minimal, .dashboard: return false
        default: return true
        }
    }

    static var `default`: HomeStyle { .editorialFull }
}

final class HomeStylePreference: ObservableObject {

    static let shared = HomeStylePreference()

    @Published var current: HomeStyle

    private let key = "speakpro.home_style"

    private init() {
        let raw = UserDefaults.standard.string(forKey: key)
        current = HomeStyle(rawValue: raw ?? "") ?? .default
    }

    func set(_ style: HomeStyle) {
        current = style
        UserDefaults.standard.set(style.rawValue, forKey: key)
    }
}
